### PR TITLE
Be less aggressive on windowInterval

### LINKED ISSUE #5070: windowInterval should never go to 0

### PR DESCRIPTION
**Pull request by bthesorceror**
- backport logic from v2 line

Closes #5070

  - [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergeable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/src/shared/parsing/flux/durations.ts [hunk 1]
```diff
@@ -197,7 +197,7 @@ function durationDuration(durationLiteral: DurationLiteral): number {
 function resolveDeclaration(ast: any, name: string): RangeCallPropertyValue {
   const isDeclarator = node => {
     return (
-      get(node, 'type') === 'VariableDeclarator' &&
+      get(node, 'type') === 'VariableAssignment' &&
```
Comment (OfTheDelmer) [inline ui/src/shared/parsing/flux/durations.ts]: Wonder how this changed?

File: ui/src/shared/parsing/flux/durations.ts [hunk 2]
```diff
@@ -197,7 +197,7 @@ function durationDuration(durationLiteral: DurationLiteral): number {
 function resolveDeclaration(ast: any, name: string): RangeCallPropertyValue {
   const isDeclarator = node => {
```
Comment (OfTheDelmer) [inline ui/src/shared/parsing/flux/durations.ts]: Worth updating this predicate and `resolveDeclaration`?

File: ui/src/dashboards/components/ReactCodeMirror.tsx [hunk 3]
```diff
@@ -334,7 +333,7 @@ class ReactCodeMirror extends PureComponent<Props, State> {
     const query = replaceTemplates(value, templates)
     const durationMs = await duration(query, config.source)
 
-    return replaceInterval(':interval:', DEFAULT_X_PIXELS, durationMs)
```
Comment (OfTheDelmer) [inline ui/src/dashboards/components/ReactCodeMirror.tsx]: :)

File: ui/src/shared/parsing/flux/durations.ts (hunk 1 above)
Comment (bthesorceror) [reply ui/src/shared/parsing/flux/durations.ts]: it changed in the flux repo as of version 0.8.0

**Pull request merged**